quickfixj-core/src/main/java/quickfix/mina/initiator/IoSessionInitiator.java

```diff
@@ -261,12 +261,13 @@ private void handleConnectException(Throwable e) {
                 e = e.getCause();
             }
             final String nextRetryMsg = " (Next retry in " + computeNextRetryConnectDelay() + " milliseconds)";
+            ConnectException wrappedException = new ConnectException(e, socketAddress);
             if (e instanceof IOException) {
                 fixSession.getLog().onErrorEvent(e.getClass().getName() + " during connection to " + socketAddress + ": " + e + nextRetryMsg);
-                fixSession.getStateListener().onConnectException(fixSession.getSessionID(), (IOException) e);
+                fixSession.getStateListener().onConnectException(fixSession.getSessionID(), wrappedException);
```

Comment:
`fixSession.getStateListener().onConnectException...` could be moved outside of the if block.